src/chapters/api/interoperability.typ

```diff
@@ -0,0 +1,105 @@
+#import "/src/util.typ": *
+#import mantys: *
+
+= Interoperability
+Ensuring interoperability between packages is an important part of API design.
+Lack of interoperability, especially when it leads to bugs or panics will generally lead users to shy away from using one of the packages which are incompatible.
+When designing the API of your package, the following two questions should be considered for every item of your package's API:
+
+- should this item be public or private?
+- should this behavior be configurable?
+
+The choice to make an item private or part of a function's behavior hardcoded can negatively impact it's interoperability, but the opposite can lead to bloated and intimidating APIs.
+Depending on the target audience of your package you may chose one over the other.
+Developers may require APIs with more configurabe parameters while end users may just want a simple to use API that doesn't get in their way.
+
+== Visibility
+In order to provide other package authors with the ability to write extensions to your package they must usually be able to access cerain internals of your package.
+The internals of your package can be exposed in different ways.
+They can be exposed as public items which are undocumented or marked as unstable.
+Or they can be exposed fully documented and stable but marked as internal or advanced.
+How and if they are exposed depends on your package's use cases and target audience.
+
+#wbox[
+  At the time of writing this Typst 0.11.1 doesn't support controlling the visiblity of items directly.
+]
+
+To mark an item private, consider simply prefixing it with an underscore like `_internal-func`.
```

Comment:
Might be worth mentioning how the `tidy` package can omit documentation for internals prefixed with an underscore, so they can still be documented in the source file without showing up in the manual. That might be too much though

Reply:
I would consider this an implementation detail of tidy, so I'll keep this out of the guidelines.

It's up to the project to decide whether such an item is documented or not.